mustache.go

```diff
@@ -565,7 +548,9 @@ func renderElement(element interface{}, contextChain []interface{}, buf io.Write
 			return err
 		}
 	case *partialElement:
-		if err := elem.tmpl.renderTemplate(contextChain, buf); err != nil {
+		if partial, err := elem.prov.Get(elem.name); err != nil {
+			return err
+		} else if err := partial.renderTemplate(contextChain, buf); err != nil {
```

Comment:
I find using the scoped variable in an else if kind of hard to read. Would prefer if this were written as:
```go
partial, err := elem.prov.Get(elem.name);
if err != nil {
	return err
}
if err := partial.renderTemplate(contextChain, buf); err != nil {
 	return err
}
```